src/sentry/api/endpoints/project_group_index.py

```diff
@@ -99,7 +102,27 @@ def get(self, request: Request, project) -> Response:
             stats_period=stats_period,
         )
 
+        hashes = request.GET.getlist("hash", [])
         query = request.GET.get("query", "").strip()
+
+        if hashes:
+            if query:
+                return Response({"detail": ERR_HASHES_AND_OTHER_QUERY}, status=400)
+
+            # limit to 100 hashes
+            hashes = hashes[:100]
+            groups_from_hashes = GroupHash.objects.filter(hash__in=hashes).values_list(
+                "group_id", flat=True
+            )
+            groups = list(Group.objects.filter(id__in=groups_from_hashes, project_id=project.id))
```

Comment:
Is there a reason to not filter on project in the original query? I'm not amazing with the Django ORM, but I think that might let you do it all in one query.

Reply:
`project` can be null on GroupHash, but not on Group I could only add the project filter on the groups after we got them

Reply:
But if the project on the grouphash is null, we don't want that record anyway. 

(It will never be the case that a grouphash has a group id but not a project id. I don't even know why `project` is nullable, since the only two places we create grouphashes ([here](https://github.com/getsentry/sentry/blob/15fcf7e673f7d7f98a6fbc3c5d430b8b1676e4a6/src/sentry/event_manager.py#L1418) and [here](https://github.com/getsentry/sentry/blob/15fcf7e673f7d7f98a6fbc3c5d430b8b1676e4a6/src/sentry/event_manager.py#L1786)) we create them with a project.)